mbed_devices/_internal/windows/device_detector.py

```diff
@@ -0,0 +1,55 @@
+"""Defines a device detector for Windows."""
+from typing import Generator
+
+# import win32com.client
+import logging
+
+from mbed_devices._internal.base_detector import DeviceDetector
+from mbed_devices._internal.candidate import Candidate
+
+
+logger = logging.getLogger(__name__)
+# https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/cim-usbdevice
+# https://docs.microsoft.com/en-gb/windows/win32/cimwin32prov/win32-usbcontroller?redirectedfrom=MSDN
+# https://docs.microsoft.com/en-gb/previous-versions/windows/desktop/cimwin32a/win32-usbhub?redirectedfrom=MSDN
+# wmi = win32com.client.GetObject ("winmgmts:")
+# for usb in wmi.InstancesOf ("Win32_USBController "):
```

Comment:
If you just need USB comms take a look at https://github.com/pyusb/pyusb which is cross platform (but does require `libusb`). Using `win32com` is a highway to insanity

Reply:
Thanks!

Reply:
@rwalton-arm Are you suggesting we could use `pyusb` instead of handling OS differences? I wasn't aware it existed. How would we go about making sure some flavor of `usblib` is shipped with `pyinstaller` / when this package is installed? 🤔

Reply:
We should be able to use `pyusb` instead of handling os differences, however the `libusb` installation would be specific to the platform. I think on windows you need to have the `libusb.dll` located in the package dir, on macOS and Linux you'd have to install `libusb` with a package manager or build from source.

Reply:
FWIW, documenting discussion offline for posterity. Points raised:
- the biggest advantage of libusb is a unified way of listing usb connected devices - we don't have to worry about os differences
- we still need to read filesystem from the device, pyusb does not make it any easier
- we might have potential issues with shipping libusb as part of the installer
- libusb is LGPL

Reply:
Thanks for the write up. I will copy it to an ADR